h2o-core/src/main/java/water/jdbc/SQLManager.java

```diff
@@ -179,27 +192,39 @@
     final double rows_per_chunk = chunk_size; //why not numRow * chunk_size / totSize; it's supposed to be rows per chunk, not the byte size
     final int num_chunks = Vec.nChunksFor(numRow, (int) Math.ceil(Math.log1p(rows_per_chunk)), false);
 
-    if (optimize) {
+    if (streaming && optimize) {
+      Log.warn("Parameter 'optimize' will be ignored in streaming ingest.");
+    }
+    final boolean estimateConcurrentConnections = optimize && (! streaming);
```

Comment:
@michalkurka should we think about merging those 2 params?
optimize is on by default anyway, and I believe that streaming is sub-optimal in most cases, except for smaller datasets, for which the optimize=false may also be slightly better that optimize=true

Reply:
besides, maybe I'm missing sth, but `streaming` looks misleading for end-user.
If we're "streaming" data internally and creating chunks on the fly, the end user can only access the fully loaded Frame, right?

Reply:
Streaming might be a wrong name - it is more of a compatibility mode. I didn't want to call it a compatibility mode because it doesn't tell the whole story. I agree with you that "streaming" is a bit confusing because it describes internal behavior in H2O. I think eventually I would like to modify the implementation in a way it doesn't need to run the COUNT query first and flip it into a truly streaming mode (similar to HTTP ingest).

Regarding combining these 2 into one - I think they serve a different purpose. We can deprecate "optimize" but I don't "streaming" should be controlled by the optimize flag (whatever that flag means) ;)

What name would you suggest?

Reply:
agree that 'optimize' is not a good name either, but still like the idea to merge them.
Today, optimize ensure the data load is "optimally" distributed, and streaming is doing the opposite, loading all the data on one node...
so, what about 'distributed' ? or sth around this?

Reply:
The data still end up distributed on all nodes. The difference is that one of the nodes reads them from the database and streams them to the rest of the cluster.

I wouldn't think optimize=False should trigger the "streaming" behavior - that is confusing to the existing users IMHO. If optimize is useless, we should deprecate it and introduce a new parameter for the new mode.

Reply:
and if you plan to switch have a true streaming mode, what about a "mode" enum instead? will make the API easier to extend, today DISTRIBUTED, EAGER, tmrw add STREAMING.
can't find an opposite for distributed :)

Reply:
The "true streaming" will be a modification of the existing code. It won't be a user-facing change. It will just make things better without users knowing :)

I do like the idea of having an enum though. That could solve the parameter name and it is extensible in the future if necessary.

Reply:
fetch_mode = distributed (default), single, streaming (for later) ?

Reply:
thanks, good idea

Reply:
"If optimize is useless, we should deprecate it and introduce a new parameter for the new mode."
agree, just thinking about using this param for this.
Internally you could do
if (optimize == true) fetch_mode = distributed
and that's it, you just need to add a @deprecated annotation to optimize flag